easybuild/easyconfigs/c/Catch2/Catch2-2.11.0-GCCcore-8.2.0.eb

```diff
@@ -0,0 +1,31 @@
+easyblock = 'CMakeMake'
+
+name = 'Catch2'
+version = '2.11.0'
+
+homepage = 'https://github.com/catchorg/Catch2'
+description = """A modern, C++-native, header-only,
+ test framework for unit-tests, TDD and BDD
+ - using C++11, C++14, C++17 and later
+   (or C++03 on the Catch1.x branch)
+"""
+
+toolchain = {'name': 'GCCcore', 'version': '8.2.0'}
```

Comment:
@boegel Why this and not `SYSTEM`?

Reply:
Because it needs `cmake` to install...

We're sometimes running into problems with `Eigen` which also requires `cmake` to install, using `SYSTEM` requires us to rely on the system compiler to install `CMake` (& its dependencies).

Reply:
Okay. Is it worth noting this as a comment?

Reply:
So, we have a couple of options here:

1) stick to using a non-system toolchain (current situation), so we can easily use a `CMake` installed with a non-system toolchain as a build dep
2) switch to using `SYSTEM` as toolchain, and using a `CMake` built with the `SYSTEM` toolchain as dependency (which is troublesome going forward since we're relying on the system compiler to build `CMake` and it's dependencies)
3) use `SYSTEM` as toolchain for `Catch2, but use a `CMake` built with `GCCcore` (which is what we're currently doing for `Eigen`, see #8261)

1) seems to be the best way w.r.t. making sure things don't break, but it's of course silly to install a header-only library using a non-system toolchain.

We've learned the hard way that 2) isn't a good option.

3) is sort of a middle ground, but it doesn't seem optimal either. For `Eigen` for example, we currently use a `CMake` built with `GCCcore/7.3.0`, which is getting a bit old...

Thoughts?

Reply:
If it wasn't for the CMake files it installes a simple copy if the files in single_include/, docs and a simple sed for correct values of the pkgconfig file would be enough.
Installing the CMake files can be done with a bit more work, so technically one doesn't need CMake to install this at all.

Reply:
The difficult one is Catch2Targets.cmake

Reply:
I'd go with the Eigen approach of having a ('CMake', '3.12.1', '-GCCcore-7.3.0') builddep, and update it from time to time to a newer GCCcore/CMake version, even for old versions of Catch2 and Eigen.

Reply:
Yeah, I guess that's the best way forward.

Changed in 576def5

Reply:
Also updated `Eigen` 3.3.7 easyconfig in #9398 (it's not worth updating the older ones imho)